docker/pebble-challtestsrv/windows.Dockerfile

```diff
@@ -0,0 +1,17 @@
+FROM golang:1.11-nanoserver-sac2016 as builder
+
+ENV CGO_ENABLED=0 GOFLAGS=-mod=vendor
+
+WORKDIR /pebble-src
+COPY . .
+
+RUN go install -v ./cmd/pebble-challtestsrv/...
+
+## main
+FROM mcr.microsoft.com/windows/nanoserver:sac2016
+
+COPY --from=builder /gopath/bin/pebble-challtestsrv.exe /gopath/bin/pebble-challtestsrv.exe
+
+RUN powershell.exe -Command $path = $env:path + ';c:\gopath\bin'; Set-ItemProperty -Path 'HKLM:\SYSTEM\CurrentControlSet\Control\Session Manager\Environment\' -Name Path -Value $path
+
+CMD [ "/pebble-challtestsrv" ]
```

Comment:
It's possible I'm not understanding the `RUN` invocation and Windows filesystem mechanics but does this `"/pebble-challtestsrv"` need to be `"/pebble-challtestsrv.exe"`?

Reply:
In Windows, file extension is more closer to a metadata about the file type, than a part of the file name. In particular for `.exe` files, it instructs Windows that the file is an executable: then it can be called with or without the `.exe` extension with the exact same behavior.

Reply:
That makes sense!

For my own curiosity could you help me understand the `RUN powershell.exe` line and how `/pebble-challtestsrv` ends up being resolved to an absolute path? It looks like the powershell fragment is required to add the `$GOPATH/bin` to the system path, but what does the registry key being set accomplish? 

Thanks for schooling me on all this Windows stuff :-)

Reply:
Again, this is because Windows is quite complicated about this. Here, this line with powershell does two things:
* an equivalent of classical Linux operation `PATH=/some/path/:$PATH` to make immediately available a new path resolution on the current shell
* an insertion in the registry to make this updated PATH modification available to any new process

The second step is necessary because in Windows, calling a shell from another shell does not use a fork approach, that would end to a complete copy of the environment (including PATH) from the parent process to the child process. It uses a thread approach: in this case, the child shell is a new thread, with an independent fresh new environment, so without the updated PATH. The fresh environment is constructed from the registry, so it gets the updated PATH from it thanks to the second operation.